crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs

```diff
@@ -807,6 +828,33 @@ pub struct MqttAuthClientConfig {
     pub key_file: Utf8PathBuf,
 }
 
+impl TEdgeConfigReaderHttpClientAuth {
```

Comment:
I still have to understand how `tedge config` works under the hood!

How is this `impl TEdgeConfigReaderHttpClientAuth` connected to `tedge_config.http.client.auth`?

Reply:
The top level tedge config that we interact with is `TEdgeConfigReader`. This struct contains a struct for each config field, with the struct names generated by appending the field name to `TEdgeConfigReader` (so `tedge_config.http` has type `TEdgeConfigReaderHttp`), and this continues recursively until a field has a value.

Because the data is stored within `http.client.auth`, I have implemented this method on `TEdgeConfigReaderHttpClientAuth` so it can be used as `tedge_config.http.client.auth.identity()`.

Reply:
Thanks. It's clearer.